cakechat/dialog_model/model.py

```diff
@@ -583,8 +584,10 @@ def print_matrices_weights(self):
         total_network_size = 0
         for p, v in zip(params, values):
             param_size = float(v.nbytes) / 1024 / 1024
+            # Work around numpy/python 3 regression: 
+            # http://www.markhneedham.com/blog/2017/11/19/python-3-typeerror-unsupported-format-string-passed-to-numpy-ndarray-__format__/
             laconic_logger.info('\t{0:<40} dtype: {1:<10} shape: {2:<12} size: {3:<.2f}M'.format(
-                p.name, v.dtype, v.shape, param_size))
+                str(p.name), str(v.dtype), str(v.shape), float(param_size)))
```

Comment:
Do you really need to convert all the format params with str(...)? I can't see any array in here that could require this conversion

Reply:
repr() or str() is required for numpy.dtype and tuple (shape) if they need to be truncated to a certain character count using `:<num` on python 3.